### PR TITLE
Game logic for validate

### PR DESCRIPTION
**Pull request by digomes87**

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    @cell[0][0] == @cell[1][0] && @cell[1][0] == @cell[2][0] ||
+      @cell[0][1] == @cell[1][1] && @cell[1][1] == @cell[2][1] ||
+      @cell[0][2] == @cell[1][2] && @cell[1][2] == @cell[2][2]
+  end
+
+  def horizontal
+    @cell[0][0] == @cell[0][1] && @cell[0][1] == @cell[0][2] ||
+      @cell[1][0] == @cell[1][1] && @cell[1][1] == @cell[1][2] ||
+      @cell[2][0] == @cell[2][1] && @cell[2][1] == @cell[2][2]
+  end
```
Comment (maelfosso) [inline lib/game.rb]: By having Baord as attribute of Game, these three methods move in Board class

File: lib/game.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+end
```
Comment (maelfosso) [inline lib/game.rb]: Why that  class ??

File: lib/players.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+class Player
+  attr_accessor :name, :element, :score
+
+  def initialize(name)
+    @name = name
+    @element = nil
+    @score = 0
+  end
+end
```
Comment (maelfosso) [inline lib/players.rb]: Great !!!

File: bin/main.rb [hunk 4]
```diff
@@ -1,3 +1,122 @@
...
+end
+
+players = info_players
+player1 = players[0]
+player2 = players[1]
+
+loop do
+  board = Board.new
+  cell = board.cell
+  display(cell)
```
Comment (maelfosso) [inline bin/main.rb]: You can't have the board out of the Game.
The Board can't exists without the Game.

File: bin/main.rb [hunk 5]
```diff
@@ -1,3 +1,122 @@
...
+
+  loop do
+    begin
+      p "Ok, #{player1.name}, it's your turn, select a number between 1-9"
+        place = gets.chomp
+        result = board.move(player1.element, place.to_i)
+        display(result[2])
+      rescue StandardError
+        p "Oh no! You have to select a number between 1-9 \nRemember to fill a blank space."
+        retry
```
[the remark this answers is not in the document]
Comment (maelfosso) [reply bin/main.rb]: Fixed !! :+1:

File: lib/board.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require_relative 'players.rb'
+require_relative 'game.rb'
+
+class Board
+  attr_accessor :cell, :counter
```
[the remark this answers is not in the document]
Comment (maelfosso) [reply lib/board.rb]: Not fixed :disappointed:

[the remark this answers is not in the document]
Comment (digomes87) [reply lib/board.rb]: what I should to do here ?

File: lib/board.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    if place <= 3
+      @cell[0][@cell[0].index(place)] = element
+    elsif place <= 6
+      @cell[1][@cell[1].index(place)] = element
+    else
+      @cell[2][@cell[2].index(place)] = element
+    end
+    result = Game.new(@cell)
+    result.win_game
+    @couter += 1
```
[the remark this answers is not in the document]
Comment (digomes87) [reply lib/board.rb]: My code runner, the game runner.
I don't understood, where stay my error ?

Screen Shot 2019-10-29 at 18 15 32

File: bin/main.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (digomes87) [reply bin/main.rb]: ???

File: lib/board.rb (hunk 7 above)
[the remark this answers is not in the document]
Comment (maelfosso) [reply lib/board.rb]: If you read when the comment has been written, you'll notice that it was 2 days ago at the first review. 
You fix it to make your code runnable.

File: bin/main.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (maelfosso) [reply bin/main.rb]: If you read the last review, the problem here is the composition principles of OOP.
The Board cannot exist outside of the Game.

**Pull request merged**